### PR TITLE
docs: Update and improve `stackablectl` documentation

### PR DESCRIPTION
**Pull request by Techassi**
Tracked by #98 

```[tasklist]
## Copy Tasks
- [X] Automatic help text generation for commands
- [X] Copy over command documentation
- [X] Copy over index page
- [x] Copy over demo pages
- [x] Copy over quick start guide
- [x] Copy over customization pages
- [x] Copy over installation and quick start pages
```

```[tasklist]
## Update Tasks
- [x] Update command texts
- [x] Update demo texts
- [x] Update customization texts
- [x] Update installation and quick start texts
```

```[tasklist]
## Repo Tasks
- [ ] https://github.com/stackabletech/documentation/pull/449
- [ ] https://github.com/stackabletech/documentation-ui/pull/47
- [ ] Update `demos` repo
```

```[tasklist]
## Operator Tasks
- [ ] https://github.com/stackabletech/airflow-operator/pull/327
- [ ] https://github.com/stackabletech/druid-operator/pull/472
- [ ] https://github.com/stackabletech/hbase-operator/pull/397
- [ ] https://github.com/stackabletech/hdfs-operator/pull/393
- [ ] https://github.com/stackabletech/hive-operator/pull/372
- [ ] https://github.com/stackabletech/kafka-operator/pull/623
- [ ] https://github.com/stackabletech/nifi-operator/pull/508
- [ ] https://github.com/stackabletech/opa-operator/pull/476
- [ ] https://github.com/stackabletech/spark-k8s-operator/pull/285
- [ ] https://github.com/stackabletech/trino-operator/pull/476
- [ ] https://github.com/stackabletech/zookeeper-operator/pull/728
```

### CODE REVIEW

File: Cargo.toml
Comment (sbernauer) [inline Cargo.toml:37]: Any particular reason to switch back to `rustls-tls-native-roots`? Or is this due to git merge stuff going on?

Comment (Techassi) [reply Cargo.toml:37]: Oh, my bad. I applied the wrong changes when merging. Will update asap.

Comment (Techassi) [reply Cargo.toml]: Reverted in 8e2100f

Comment (sbernauer) [reply Cargo.toml]: Awesome, thanks!

**Comment by Techassi** [comment]
The content adjustments are taking longer than expected. Working through the text requires fixing grammar errors and simplifying sentences. Also, many longer links are converted to references, so that the text flows better when these links are used.

File: docs/modules/stackablectl/pages/commands/cache.adoc
Comment (fhennig) [inline docs/modules/stackablectl/pages/commands/cache.adoc:2]: Explain a bit more what is actually cached and why. Also maybe where?

File: docs/modules/stackablectl/pages/commands/completions.adoc
Comment (fhennig) [inline docs/modules/stackablectl/pages/commands/completions.adoc:2]: Add something like "This command is only used to generate shell completions and normally doesn't need to be used by the user"

**Review by fhennig — COMMENTED** [review]
I like the generated stuff. How can we make sure that the generated content gets updated if the code is changed?

File: docs/modules/stackablectl/pages/commands/cache.adoc
Comment (Techassi) [reply docs/modules/stackablectl/pages/commands/cache.adoc:2]: Will be added in a follow-up PR.

File: docs/modules/stackablectl/pages/commands/completions.adoc
Comment (Techassi) [reply docs/modules/stackablectl/pages/commands/completions.adoc:2]: Will be added in a follow-up PR.

**Review by fhennig — APPROVED** [review]

**Comment by Techassi** [comment]
> I like the generated stuff. How can we make sure that the generated content gets updated if the code is changed?

This is generated by a pre-commit hook which uses `cargo xtask gen-ctl-readme`.

**Comment by lfrancke** [comment]
A few items are left unchecked/unmerged in the tasklist. What's the status there?

**Comment by Techassi** [comment]
They are still ongoing and we are continuing to use this PR for tracking them. However, this PR cannot be re-opened after it was merged.

**Comment by lfrancke** [comment]
Okay, but then it shouldn't be in the `Done` column. 
We usually use issues for that and not PRs. It's a bit weird but let's then move it back.

**Comment by Techassi** [comment]
Agreed, it's a little weird. This PR started small, but spiraled a little out of control. Changes here required more and more changes in other repositories, so that's why I started tracking those tasks here.

**Comment by lfrancke** [comment]
The other alternative is to create a new issue but I'll leave that to you.

**Comment by Techassi** [comment]
Yeah we are pretty much done. So it should be fine. But in the future it might be better to track this using an issue.

**Pull request merged**
